web/impact/impact/tests/test_accelerator_routes.py

```diff
@@ -0,0 +1,174 @@
+
+    def test_api_object_list(self):
+        StartupFactory(is_visible=1)
+        StartupFactory(is_visible=1)
+        StartupFactory(is_visible=1)
+        url_name = "object-list"
+        view_kwargs = {'app': 'accelerator', "model": "startup"}
+        self.response_401(self.get(url_name, **view_kwargs))
+
+        basic_user = self.make_user('basic_user@test.com')
+        with self.login(basic_user):
+            self.response_403(self.get(url_name, **view_kwargs))
+        startup_permission, _ = Permission.objects.get_or_create(
+            content_type=ContentType.objects.get(
+                app_label='mc',
+                model='startup'),
+            codename='view_startup',
+        )
+        perm_user = self.make_user(
+            'perm_user2@test.com', perms=["mc.view_startup"])
+        perm = PermissionFactory.create(codename='change_startup')
+        view_perm = PermissionFactory.create(codename='view_startup')
+        perm_user.user_permissions.add(perm)
+        perm_user.user_permissions.add(startup_permission)
+        perm_user.user_permissions.add(view_perm)
+        perm_user.save()
+        with self.login(perm_user):
+            response = self.get(url_name, **view_kwargs)
+            self.response_200(response)
+            response_dict = json.loads(response.content)
+            self.assertIn("short_pitch", response_dict['results'][0].keys())
+            for startup in response_dict['results']:
+                self.assertEqual(startup["is_visible"], 1)
+
+    def test_api_object_get(self):
+        url_name = "object-detail"
+        industry = IndustryFactory()
+        StartupFactory(id=1, primary_industry_id=industry.id)
+        view_kwargs = {
+            'app': 'accelerator',
+            "model": "startup",
+            "pk": 1,
+        }
+        self.response_401(self.get(url_name, **view_kwargs))
+        startup_permission, _ = Permission.objects.get_or_create(
+            content_type=ContentType.objects.get(
+                app_label='mc',
+                model='startup'),
+            codename='view_startup',
+        )
+        basic_user = self.make_user('basic_user@test.com')
+        with self.login(basic_user):
+            self.response_403(self.get(url_name, **view_kwargs))
+
+        perm_user = self.make_user(
+            'perm_user@test.com', perms=["mc.view_startup"])
+        perm = PermissionFactory.create(codename='change_startup')
+        view_perm = PermissionFactory.create(codename='view_startup')
+        perm_user.user_permissions.add(perm)
+        perm_user.user_permissions.add(startup_permission)
+        perm_user.user_permissions.add(view_perm)
+        perm_user.save()
+        with self.login(perm_user):
+            response = self.get(url_name, **view_kwargs)
+            self.response_200(response)
+            response_dict = json.loads(response.content)
+            self.assertIn("is_visible", response_dict.keys())
```

Comment:
I was expecting a test that involved the `IndustryFactory()` shown in the first act.. Is there a reason it's missing? add an assert or remove the misleading setup.

Reply:
added an assert to lock this down - the industryfactory was needed because primary industry for startup is a required field that needs a value.

Reply:
A Context would've also worked here well, but this is a reasonable alternative for now.